### PR TITLE
Introduce EnvironmentAdapter in place of action and state mapper functions

### PR DESCRIPTION
**Pull request by smiszym**
We already had action and state conversion functions for conversions
between the environment representation and the ACS2 representation.
Whenever the environment representation of actions or states was
incompatible with what ACS2 received and produced, adapter functions
needed to be used.

However, we only had uni-directional converters:
  * env state -> ACS2 state
  * ACS2 action -> env action

The goal of this commit is to add the reverse converters, i.e.:
  * env action -> ACS2 action
  * ACS2 state -> env state
so that, e.g., `Classifier.__str__()` can be implemented properly.

For instance, this allows `Classifier.__str__()` to print `N`, `W`, `SE`,
etc. in the action part instead of `0`, `6`, `3`, etc. if the proper
environment adapter is provided.

Notebooks not updated (yet).

RACS not touched.

### CODE REVIEW

**Comment by smiszym** [comment]
To get the idea, start reading from EnvironmentAdapter.py and the example subclass in acs2_in_multiplexer.py

What do you think about it? Too complicated and better off just adding two reverse mapper functions next to the existing ones in Configuration? :)

File: lcs/agents/acs2/EnvironmentAdapter.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    @staticmethod
+    def env_state_to_acs(env_state):
+        """
+        Converts environment representation of a state to ACS2
+        representation.
+        """
+        return env_state
+
+    @staticmethod
+    def acs_action_to_env(acs_action):
```
Comment (khozzy) [inline lcs/agents/acs2/EnvironmentAdapter.py]: I wouldn't put `acs` inside function name (make it more general). Maybe just `to_env_action(lcs_action)`

File: lcs/agents/acs2/EnvironmentAdapter.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    This implementation works for environments which provide ACS2-compatible
+    state and action representations (no conversion needed).
+
+    Subclass this class in pyALCS integration to provide an adapter
+    for a specific environment.
+    """
+
+    @staticmethod
+    def env_action_to_acs(env_action):
```
Comment (khozzy) [inline lcs/agents/acs2/EnvironmentAdapter.py]: `to_lcs_action(env_action)`

**Comment by khozzy** [comment]
I like this idea. But would prefer to use some more LCS specific terminology. Your changes are also a good first step towards #33

**Comment by smiszym** [comment]
Thanks for renaming uses of this functions, I forgot that.

**Pull request merged**